### PR TITLE
{lang} [intel/2016b] LLVM 3.9.0

### PR DESCRIPTION
**Pull request by wpoely86**

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86 please make this `(WIP)` since LLVM is not there yet

File: easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-foss-2016b.eb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+dependencies = [
+    ('ncurses', '6.0'),
+    ('zlib', '1.2.8'),
+]
+
+configopts = '-DBUILD_SHARED_LIBS=ON '
+# required to install extra tools in bin/
+configopts += '-DLLVM_INSTALL_UTILS=ON -DLLVM_TARGETS_TO_BUILD=X86 -DLLVM_ENABLE_ZLIB=ON '
+configopts += '-DCMAKE_BUILD_TYPE=Release '
```
Comment (boegel) [inline easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-foss-2016b.eb]: @wpoely86 isn't this the default?

File: easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-intel-fix-loop.patch [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Bug in intel compiler
```
Comment (boegel) [inline easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-intel-fix-loop.patch]: actually, it seems like it's a bug in LLVM not conforming to the C++ standard, see https://software.intel.com/en-us/forums/intel-c-compiler/topic/640702

File: easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-foss-2016b.eb (hunk 1 above)
Comment (wpoely86) [inline easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-foss-2016b.eb]: nope

File: easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-intel-fix-loop.patch (hunk 2 above)
Comment (wpoely86) [inline easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-intel-fix-loop.patch]: no, it's icpc not sticking to the standard.

Comment (boegel) [inline easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-intel-fix-loop.patch]: ah, ok, my bad

File: easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-foss-2016b.eb (hunk 1 above)
Comment (boegel) [inline easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-foss-2016b.eb]: wait, so has this changed then? or do we need to make this same change in other LLVM easyconfigs (in a separate PR)?

Comment (wpoely86) [inline easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-foss-2016b.eb]: Probably have to change it in the other easyconfigs too.

Comment (boegel) [inline easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-foss-2016b.eb]: is there any way we can check with a sanity check command?

Comment (wpoely86) [inline easybuild/easyconfigs/l/LLVM/LLVM-3.9.0-foss-2016b.eb]: no idea. You could check for debugging symbols in the binary, but that doesn't necessarily imply a debug build.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
build2901.metapod.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU           X5675  @ 3.07GHz, Python 2.7.5
See https://gist.github.com/9a818df62361779bb2d8e28952199fb0 for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 34 out of 35 (3 easyconfigs in this PR)
node2602.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/f9989e9b6ae9d4cbc0a0e5fc5a1e948d for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 34 out of 35 (3 easyconfigs in this PR)
node2601.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/1db281c4cc8fcc8893fc41580ffddc2a for a full test report.

**Comment by wpoely86** [comment]
Test report by @wpoely86
**FAILED**
Build succeeded for 4 out of 5 (3 easyconfigs in this PR)
node2007.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/93e75ea41896100e8ab3b3d58f68e568 for a full test report.

**Comment by wpoely86** [comment]
@boegel ready for merging. I've currently dropped Clang as it has weird build issues and I only need LLVM for now. I will put in a seperate PR.

**Comment by boegel** [comment]
lgtm, but we should open a PR for the `Release` target configure option too in the other LLVM easyconfigs...

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 3 (2 easyconfigs in this PR)
node2003.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/27d18179f743fdf20c577a1508e4d558 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 1 out of 3 (2 easyconfigs in this PR)
node2533.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/006d9e8696847d36d35ae06e3023a5a6 for a full test report.

**Comment by boegel** [comment]
@wpoely86 it needs to work before we can merge it, of course ;)

**Comment by wpoely86** [comment]
As I said, the tarballs are only coming tomorrow or so. 

And why are you retesting it? I'm not gonna bother sending test reports if you retest it anyway.

**Comment by boegel** [comment]
@wpoely86 I have a script to merge PRs, which verifies a couple of things before actually merging the PR. One requirement is that the most recent test report is a successful one, which was not the case here, so I submitted a test job for this PR myself (well, two).

You mentioned this PR is ready to be merged, so I assumed the upstream sources were available. 
Apparently, they're not, so the PR is not good to go.

Please only marked the PR as ready-to-go when the sources are available from the specified source URLs. If the (official) source tarball is available from another location earlier, we should reconsider switching to it?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2018.delcatty.os - Linux SL 6.8, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/c76b33af1c77d0aaba6277279215af52 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2554.golett.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/5288df861ea39febb12bc2b6d2691e35 for a full test report.

**Comment by boegel** [comment]
lgtm

**Comment by boegel** [comment]
Going in, thanks @wpoely86!

**Pull request merged**